### DISCUSSION TITLE
Modal Usage Question

### DISCUSSION THREAD

**Discussion by ploring04** [DSharpPlus/DSharpPlus#1220] [Q&A]
Thanks for adding the basic modal support so quickly!

I have a slashcommand /registeruser which then pops up a modal with the fields needed which works fine.  

I am then trying to do a WaitForModalAsync() to get the values before continuing, but since I did a CreateResponseAsync(InteractionResponseType.Modal,...)  to display the modal I seem to be having a problem since the interaction didn't send a response within 3 seconds.

So something like this:

```
        [SlashCommand("register", "Register Player")]
        public async Task RegisterPlayerCommand(InteractionContext ctx)
        {
            try
            {
                var input = ctx.Client.GetInteractivity();

                var modal= new DiscordInteractionResponseBuilder();
                modal.WithTitle("Register Player")
                              .WithCustomId("registerModal")
                              .AddComponents(new TextInputComponent("Player Name", "playerName", required: true, min_length: 1, max_length: 64))
                              .AddComponents(new TextInputComponent("Location", "location", required: false));

                await ctx.CreateResponseAsync(InteractionResponseType.Modal, modal);

                var result = await input.WaitForModalAsync("registerModal");
                // Process results...
            }
            catch (Exception e)
            {
            }
        }
```

Is that the correct flow of using a modal?

**Comment by JulianusIV** [comment]
I am unsure of what your actual problem is here, so its hard to tell what is going wrong, without any code or screenshots of what is happening.

**Reply by ploring04** [reply]
Good point!  Edited original question with a code example.

**Comment by VelvetToroyashi** [comment]
Submitting a modal invokes an interaction and you must create a response to it like you would any other interaction. Type 6 (DeferredMessageUpdate) *should* work. 

Other than that, this appears to be the "correct flow".

**Reply by ploring04** [reply]
Thanks!  That's what I was missing!

Added this for the ModalSubmitted event and working as expected:

```
        private async Task Client_ModalSubmitted(DiscordClient sender, DSharpPlus.EventArgs.ModalSubmitEventArgs e)
        {
            await e.Interaction.CreateResponseAsync(InteractionResponseType.DeferredMessageUpdate);
            Client.Logger.LogInformation($"Client_ModalSubmitted: {string.Join(",",e.Values)}");
        }
```
